### ISSUE TITLE
Possible issue with integer keys + heterogeneous siblings

### ISSUE THREAD

**Issue by steven88888** [developit/preact#38]
One component named Todo has a render function as following. It will ganerate one Todo list with js codes compiled  by React Templates. 2 problems will happen:
1. If todo.key is integer (which can be zero),  add operation would be not work well. 
2. After delete one todo array element  of state,  UI element can not be removed correspondingly.

I am a fresher of Preact. Maybe they are related to  preact/src/vdom/build.js, please check and confirm. Thanks.

```js
<div> 
    <div key="{todo.key}" class="todo-item" rt-repeat="todo in this.state.todos">        
        <img src="src/assets/delete.png"
             onClick="()=>this.remove(todo)"
             title="Remove Todo" />
        <input type="checkbox" checked="{todo.done}" onChange="()=>this.toggleChecked(todoIndex)"/>
        <span rt-class="{done: todo.done}">{todo.value}</span>
    </div>  
    <input key="myinput" class="new-todo" placeholder="What needs to be done?" type="text"
           value="{this.state.edited}"
           onKeyDown="{(e) => this.inputKeyDown(e)}"
           onInput="{this.linkState('edited')}"
           autoFocus="{true}"/>
    <button onClick="()=>this.add()">Add</button><br/>
    <button onClick="()=>this.clearDone()">Clear done</button>
</div>
```

**Comment by steven88888** [comment]
If i remove the attribute key of <div>,   the 2nd problem will disappear.

**Comment by developit** [comment]
Hi Steven,

I've never used React Templates, and honestly I wouldn't recommend an approach like that in the first place.  Here's a working example of what you wrote, without any of the templating stuff:

https://jsfiddle.net/developit/hq33t3vm/

<img width="189" src="https://cloud.githubusercontent.com/assets/105127/12818666/c9e1e910-cb25-11e5-97f3-212086c5f0e5.png">

That being said, it looks like there is an issue with integer keys, which I will certainly look into!

**Comment by steven88888** [comment]
Hi Jason, 
thanks for your answer.

**Comment by developit** [comment]
Fixed in `3.4.0`!
